hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractSoftwareModuleDetails.java

```diff
@@ -0,0 +1,206 @@
+        super(i18n, eventBus, permissionChecker, managementUIState);
+        this.softwareModuleAddUpdateWindow = softwareModuleAddUpdateWindow;
+        this.softwareManagement = softwareManagement;
+        this.swMetadataPopupLayout = swMetadataPopupLayout;
+
+        swmMetadataTable = new SoftwareModuleMetadatadetailslayout();
+        swmMetadataTable.init(getI18n(), getPermissionChecker(), softwareManagement, swMetadataPopupLayout,
+                entityFactory);
+
+        addTabs(detailsTab);
+    }
+
+    /**
+     * MetadataEvent.
+     *
+     * @param event
+     *            as instance of {@link MetadataEvent}
+     */
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEvent(final MetadataEvent event) {
+        UI.getCurrent().access(() -> {
+            final MetaData softwareModuleMetadata = event.getMetaData();
+            if (softwareModuleMetadata != null && isSoftwareModuleSelected(event.getModule())) {
+                if (event.getMetadataUIEvent() == MetadataEvent.MetadataUIEvent.CREATE_SOFTWARE_MODULE_METADATA) {
+                    swmMetadataTable.createMetadata(event.getMetaData().getKey());
+                } else if (event
+                        .getMetadataUIEvent() == MetadataEvent.MetadataUIEvent.DELETE_SOFTWARE_MODULE_METADATA) {
+                    swmMetadataTable.deleteMetadata(event.getMetaData().getKey());
+                }
+            }
+        });
+    }
+
+    @EventBusListenerMethod(scope = EventScope.UI)
+    void onEvent(final SoftwareModuleEvent softwareModuleEvent) {
+        onBaseEntityEvent(softwareModuleEvent);
+    }
+
+    @Override
+    protected void addTabs(final TabSheet detailsTab) {
+        detailsTab.addTab(createDetailsLayout(), getI18n().get("caption.tab.details"), null);
+        detailsTab.addTab(createDescriptionLayout(), getI18n().get("caption.tab.description"), null);
+        detailsTab.addTab(createLogLayout(), getI18n().get("caption.logs.tab"), null);
+        detailsTab.addTab(swmMetadataTable, getI18n().get("caption.metadata"), null);
+    }
+
+    @Override
+    protected void populateMetadataDetails() {
+        swmMetadataTable.populateSMMetadata(getSelectedBaseEntity());
+    }
+
+    @Override
+    protected void onEdit(final ClickEvent event) {
+        final Window addSoftwareModule = softwareModuleAddUpdateWindow
+                .createUpdateSoftwareModuleWindow(getSelectedBaseEntityId());
+        addSoftwareModule.setCaption(getI18n().get("upload.caption.update.swmodule"));
+        UI.getCurrent().addWindow(addSoftwareModule);
+        addSoftwareModule.setVisible(Boolean.TRUE);
+    }
+
+    @Override
+    protected String getEditButtonId() {
+        return UIComponentIdProvider.UPLOAD_SW_MODULE_EDIT_BUTTON;
+    }
+
+    @Override
+    protected String getDefaultCaption() {
+        return getI18n().get("upload.swModuleTable.header");
```

Comment:
In general we should create constants for these message keys and do this in a separate task.